server/blob/blob-cassandra/src/main/java/org/apache/james/blob/cassandra/cache/CachedBlobStore.java

```diff
@@ -237,10 +251,16 @@ private boolean isAbleToCache(byte[] bytes) {
     }
 
     private Mono<byte[]> readFromCache(BlobId blobId) {
-        return Mono.from(cache.read(blobId));
+        return Mono.from(metricFactory.runPublishingTimerMetric(BLOBSTORE_CACHED_LATENCY_METRIC_NAME, cache.read(blobId)))
+            .doOnNext(bytes -> metricRetrieveHitCount.increment());
     }
 
     private Mono<byte[]> readBytesFromBackend(BucketName bucketName, BlobId blobId) {
         return Mono.from(backend.readBytes(bucketName, blobId));
```

Comment:
To be fully complete it would really be good IMO to have a graph about backend latencies (in addition to the cache latencies, and the composed cachedBlobStore latencies)